xonsh/history/json.py

```diff
@@ -445,6 +446,12 @@ def __init__(
             if save_cwd is not None
             else XSH.env.get("XONSH_HISTORY_SAVE_CWD", True)
         )
+        ignore_regex = XSH.env.get("XONSH_HISTORY_IGNORE_REGEX")
```

Comment:
this better be moved to the historybackend base class.

Reply:
The discussion in #3279 was explicit about not putting it in the backend. I personally don't have any stake in this, I'll put it wherever you want, but if you and @anki-code don't agree, I'm not sure what to do about that.